### DISCUSSION TITLE
ESP32-S3 DevKitC & NRF24L01+ pin_mapping

### DISCUSSION THREAD

**Discussion by ButterBetzi** [hoylabs/OpenDTU-OnBattery#1317] [Q&A]
Hi,
ich habe folgendes problem: ich bin von einem Gerneric ESP 38 Pin auf den ESP32-S3 DevKitC umgestiegen. Den wollte ich genauso nutzen wie den vorherhigen mit meinem NRF24L01+. Ich bekomme es aber zum verrecken nicht mehr ans laufen. Die Antenne wird laut WebUI erkannt und ist verbunden, aber der Wechselrichter ist nicht erreichbar. Folgendes pin_mapping habe ich:
```
{
        "name": "ESP32-S3",
        "nrf24": {
            "en": 4,
            "clk": 12,
            "miso": 13,
            "mosi": 11,
            "cs": 5,
            "irq": 3
        },
        "victron": {
            "rx": 20,
            "tx": -1
        },
        "battery": {
            "rx": 21,
            "tx": 47
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": -1,
            "clk_mode": -1
        }
    },
```
Kann mir da jemand helfen?

![grafik](https://github.com/user-attachments/assets/a24d8a38-0f13-4ecd-a692-df531e14d878)
![Sketch_Steckplatine](https://github.com/user-attachments/assets/540b80ac-e3e0-455a-b111-43ddbcbc3c47)

**Comment by Manos1966** [comment]
Hallo @ButterBetzi 

bevor ich dein Pinout angucke: 
- hast du JK BMS und Victron entfernt 
- getestet ob, nur mir NRF24+ angeschlossen 

dein ESP32-S3 eine Verbindung machen kann?

**Reply by ButterBetzi** [reply]
nope, hab ich noch nicht getestet. Mache ich eben

**Reply by ButterBetzi** [reply]
scheiße, das klappt wirklich.
Wenn ich die beiden entferne läufts.
Wieso ist das so?

**Reply by Manos1966** [reply]
Also, das Problem ist nicht der NRF24, sondern die anderen zwei Module. 👍 

Schliesse jetzt an das naechste Modul (entweder JK BMS, oder Victron) und teste bis du das Modul gefunden hast, dass das Problem verursacht.

Viel Erfolg!

**Reply by ButterBetzi** [reply]
Leider doch Fehlalarm :/
ging kurz, jetzt schon wieder nicht. Sieht für mich demnach nach nem Wackelkontakt aus. Werde mal andere Jumper kabel testen

**Reply by frank-tanne** [reply]
Stromversorgung der 3v reicht nicht aus ,der Spannungsregler des ESP32-S3 DevKit ist zu schwach.

**Reply by ButterBetzi** [reply]
@frank-tanne shit. Heißt das, ich kann das generell knicken mit dem ESP? Aber wieso ist er dann als einer der Beispiel ESPs in der doku angegeben? :/
Oder ist einfach nur mein USB Netzteil zu schwach?

**Reply by Manos1966** [reply]
> Oder ist einfach nur mein USB Netzteil zu schwach?

👍 

Ich habe keine Probleme mit meiner ESP32-S3 und ich habe NRF24, CANbus und MC2515 (3,3V+5V) laufen.

**Reply by frank-tanne** [reply]
Ich hatte bei mir das 2,42 zoll Display und NRF24 an 3,3v angeschlossen da hatte ich keinen empfang zum WR. Display abgeklemmt jede Sekunde kam ein Signal.  Jetzt habe ich einen externen Spannungsregler 5v zu 3v auf 5v geklemmt und an den 3v hängt mein Display.
Läuft super.

**Reply by schlimmchen** [reply]
> Aber wieso ist er dann als einer der Beispiel ESPs in der doku angegeben? :/

Das ist kein grundsätzliches Problem. Kann gut sein, dass du einen Elektrolytkondensator an der Spannungsversorgung des NRF24 hinzufügen musst -- auch wenn es vorher ohne ging.

**Reply by Manos1966** [reply]
> **ging kurz**, jetzt schon wieder nicht. 

diese Aussage spricht Bänder...

Also fangen wir an mit der Schnitzjagt:
Bilder Bitte! 
von deinen ESP32-S3, NRF24+, USB Netzteil.
Ist es die gleiche NRF24+ Platine die du bei deinen ESP32 benutzt hast, oder ist es eine neue?

Welche Firmware Version hast du installiert? Ich hoffe die aktuelle 2024.09.11

Hast du gleichzeitig den alten ESP32 mit NRF24 auch am laufen? (plötzlich zwei NRF24 greifen auf den gleichen Wechselrichter)

Wie weit entfernt ist deine HOYMILES?

Abfrageintervall auf 5 Sekunden stellen (oder laenger)
**Ausführliche Protokollierung EINSCHALTEN,** andere Ausführliche Protokollierungen AUSSCHALTEN
NRF24 Sendeleistung: auf Minimum stellen (ich hoffe die HOYMILES ist nicht zu weit weg)

Danach die Konsole nutzen (Info-> Konsole) und versuche zu erkennen ob Fehler zu erkennen sind.
Du kannst sogar das Ergebnis der Konsole speichern: [create a log file with Console](https://github.com/hoylabs/OpenDTU-OnBattery/wiki/How-to-Search-for-Solutions%2C-Ask-Questions%2C-Report-Problems#41-create-a-logfile-datei-with-console)


Langfristig gesehen, es wäre sinnvoll ein USB Strommessgerät zu besorgen.
Meine HUAWEI Test Installation mit ESP32-S3, NRF24,  SN65HVD230 CAN-Board, MC2515 CAN-Board verbraucht nur 0,25A
![20241012_224617](https://github.com/user-attachments/assets/22d6d3f7-9234-49b7-9962-8206d6b27d57)

**Reply by frank-tanne** [reply]
Die USB Spannungstester sind doch nur Schätzeisen . Und sagt noch lange nicht aus wieviel Strom der 3,3v Spannungsregler auf den ESP Bord mit macht , da scheint es mächtige Toleranzen zu geben.

**Reply by Manos1966** [reply]
> . Und sagt noch lange nicht aus wieviel Strom der 3,3v Spannungsregler auf den ESP Bord mit macht , da scheint es mächtige Toleranzen zu geben.

@frank-tanne Jetzt ernsthaft, du versuchst mir glaubhaft zu machen, dass der  3,3v Spannungsregler des ESP32-S3 solche "mächtige Toleranzen" hat, dass manche nicht in der Lage sind, eine NRF24+ Platine als einziger Verbraucher mit ausreichend Strom zu versorgen?

**Reply by Sorbat** [reply]
Ob als einzigen Verbraucher weiß ich nicht, aber ein Elko wird beim NRF ja ausdrücklich empfohlen. Ich hatte von so einigen Fällen gelesen, bei denen die Probleme am Ende am fehlenden Elko lagen. 
![image](https://github.com/user-attachments/assets/756b6467-2260-43b8-ae74-b0e5d9bb4977)

**Reply by frank-tanne** [reply]
> > . Und sagt noch lange nicht aus wieviel Strom der 3,3v Spannungsregler auf den ESP Bord mit macht , da scheint es mächtige Toleranzen zu geben.
> 
> @frank-tanne Jetzt ernsthaft, du versuchst mir glaubhaft zu machen, dass der 3,3v Spannungsregler des ESP32-S3 solche "mächtige Toleranzen" hat, dass manche nicht in der Lage sind, eine NRF24+ Platine als einziger Verbraucher mit ausreichend Strom zu versorgen?

Ja ich bin davon überzeugt. Nicht umsonst soll man die sendeleistung runter setzen (somit weniger Reichweite) und der Elko ist ja auch nur ein Spannungsbuffer. Im übrigen will ich niemandem was glaubhaft machen, das ist nur meine Erfahrung.

**Reply by Manos1966** [reply]
OK, OK 🤣 
obwohl ich nicht ganz-überzeugt bin, ein ELKO-Manko den **kompletten Verlust** der Verbindung verursachen könnte
Ich bin gespannt auf die Bilder, vor allem vom USB-Netzteil (und Kabel).
Vielleicht liegt es wirklich daran. Ich habe immer gute USB-Netzteile und Kabel benutzt, merke aber in letzter Zeit viele Import-USB-Netzteile die die Hälfte wiegen und trotzdem die doppelte Leistung liefern (sollen) als meine. 🙈

**Reply by ButterBetzi** [reply]
Moiiiiiiin,
ich melde mich jetzt auch wieder zurück.
Also, ich habe einfach alle Tipps auf einmal umgesetzt, da ich keine Lust auf ändern, testen, ändern, testen, ... hatte :D
Habe jetzt 
1. stärkeres Netzteil
2. hochwertigeres Kabel
3. alle jumperkabel ausgetauscht
4. 100uF Elko parallel geschaltet

und siehe da: es läuft. Aber leider nur für wenige Sekunden/Minuten.
Jetzt ist die Verbindung wieder hinüber.

Hier noch die Antworten für @Manos1966 

> Ist es die gleiche NRF24+ Platine die du bei deinen ESP32 benutzt hast, oder ist es eine neue?

jup

> Welche Firmware Version hast du installiert? Ich hoffe die aktuelle 2024.09.11

jup

> Hast du gleichzeitig den alten ESP32 mit NRF24 auch am laufen? (plötzlich zwei NRF24 greifen auf den gleichen Wechselrichter)

nope

> Wie weit entfernt ist deine HOYMILES?

schätzungsweise 30 cm

Sendeleistung ist auf minimum.
![2024-10-14 16-51-53 3585](https://github.com/user-attachments/assets/35ddc7d9-1148-45ad-8c0c-f7539a018d5b)
![2024-10-14 16-52-06 3586](https://github.com/user-attachments/assets/11dd5cfe-4aa7-450b-bfb1-f4b58398deef)
![2024-10-14 16-52-11 3587](https://github.com/user-attachments/assets/1b7e9646-254f-4cb4-9940-1478d98202c1)
![2024-10-14 16-52-21 3588](https://github.com/user-attachments/assets/85aeed3b-46ba-405b-bd05-894de497132e)
![2024-10-14 16-52-29 3589](https://github.com/user-attachments/assets/ab902f38-6e25-48a3-bb45-26f0707ad0ed)
![2024-10-14 16-53-19 3591](https://github.com/user-attachments/assets/9e162eea-cd02-41b0-99d1-bbe395a732d7)

**Reply by Manos1966** [reply]
"Aber leider nur für wenige Sekunden/Minuten."
🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 

Danke @ButterBetzi du hast alles gemacht. 💯 

OK, der NRF24 ist ein "Monster -Teil" (besser ein ELKO zu haben 🙊 ) fuer Reichweiten bis 500M!
30cm ist vieeeeel zu Nah! (trotz Minimum).
Abstand vergrossern bzw etwas zwischen NRF24 und Wechselrichter stellen (ein guter Ziegelstein macht den Job).

Obwohl ich es Dir wuensche, leider glaube ich nicht, dass es daran liegt.

Es ist Zeit, dass du die Konsole einschaltest und guckst was passiert:
Ausfuehrliche Protokolierung ueberall AUSSCHALTEN, ausser bei DTU  
Danach: Info -> Konsole
und guck ob du Probleme erkennen kannst.
Falls nicht, die Logdatei auf ww.paste.bin speichern und den Link hier posten.
Anleitung hier: https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Wie-man-nach-L%C3%B6sungen-sucht%2C-Fragen-stellt%2C-Probleme-meldet#41-logdatei-mit-hilfe-der-web-konsole-erstellen

**Reply by Sorbat** [reply]
Hallo, ich hab selbst gar kein NRF Zeug mehr, aber mir wurde vor einiger Zeit mal dieses Wiring geschickt 
![image](https://github.com/user-attachments/assets/79c61371-0ca5-4d73-9c82-8e9bfcb4e8b2)
